src/board.rs

```diff
@@ -228,6 +299,115 @@ impl Chessboard {
         // Return the FEN string
         string_array.join(" ")
     }
+
+    /* *************** */
+    /* PRIVATE FUNCTIONS */
+
+    /// Maps the pieces on the chessboard to their character representations in the console.
+    ///
+    /// # Returns
+    ///
+    /// A vector of tuples, where each tuple consists of a chess piece character and its
+    /// corresponding bitboard positions. The characters represent different chess pieces,
+    /// and the bitboard positions indicate the squares occupied by those pieces on the board.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use kn_o3::Chessboard;
+    ///
+    /// let initial_position = Chessboard::new();
+    /// let pieces_mapping = initial_position.get_pieces();
+    /// for (piece_char, positions) in pieces_mapping {
+    ///     println!("Piece: {} | Bitboard Positions: {}", piece_char, positions);
+    /// }
+    /// ```
+    ///
+    /// The function returns a vector containing tuples, each associating a chess piece
+    /// character ('P', 'N', 'B', 'K', 'Q', 'R', 'p', 'n', 'b', 'k', 'q', 'r') with its
+    /// corresponding bitboard positions on the chessboard.
+    fn get_pieces(&self) -> Vec<(char, u64)> {
+        vec![
+            ('P', self.white_pawns),
+            ('N', self.white_knights),
+            ('B', self.white_bishops),
+            ('K', self.white_king),
+            ('Q', self.white_queen),
+            ('R', self.white_rooks),
+            ('p', self.black_pawns),
+            ('n', self.black_knights),
+            ('b', self.black_bishops),
+            ('k', self.black_king),
+            ('q', self.black_queen),
+            ('r', self.black_rooks),
+        ]
+    }
+
+    /// Determines the foreground color based on the chess piece color.
+    ///
+    /// # Arguments
+    ///
+    /// - `p`: The character representation of a chess piece.
+    ///
+    /// # Returns
+    ///
+    /// The `Color` of the chess piece. If the piece is uppercase (representing a white piece),
+    /// it returns `Color::White`; otherwise, it returns `Color::Black`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use chess_display::Color;
+    /// use chess_display::ChessboardDisplay;
+    ///
+    /// let display = ChessboardDisplay::new();
+    /// let piece_color = display.find_fg('K');
+    /// println!("Color of white king: {:?}", piece_color);
+    /// ```
+    ///
+    /// The function determines the color of a chess piece based on its character representation.
+    /// Uppercase letters represent white pieces, and lowercase letters represent black pieces.
+    /// The resulting color is returned as either `Color::White` or `Color::Black`.
```

Comment:
This function is pretty self explanatory and probably doesn't need all this documentation.
Leave the arguments and return, but everything else can go.